crates/core/plugin_sm/src/operation_logs.rs

```diff
@@ -47,7 +46,11 @@ impl OperationLogs {
         Ok(operation_logs)
     }
 
-    pub async fn new_log_file(&self, kind: LogKind) -> Result<LogFile, OperationLogsError> {
+    pub async fn new_log_file(
+        &self,
+        kind: LogKind,
+        operation_name: Option<&str>,
+    ) -> Result<LogFile, OperationLogsError> {
```

Comment:
This signature would be much simpler and the 2 existing `LogType` variants can be replaced with their String equivalents.
```suggestion
    pub async fn new_log_file(
        &self,
        operation_name: Option<&str>,
    ) -> Result<LogFile, OperationLogsError> {
```